alf/algorithms/agent.py

```diff
@@ -164,7 +172,32 @@ def rollout(self,
         new_state = new_state._replace(rl=rl_step.state)
         info = info._replace(rl=rl_step.info)
 
-        # TODO: can avoid computing this when collecting exps
+        # TODO
+        # avoid computing this when collecting exps (off policy train)
+        if self._entropy_target_algorithm:
+            et_step = self._entropy_target_algorithm.train_step(rl_step.action)
+            info = info._replace(entropy_target=et_step.info)
+
+        return PolicyStep(action=rl_step.action, state=new_state, info=info)
+
+    def train_step(self, exp: Experience, state):
+        new_state = AgentState()
```

Comment:
Do you need to first convert Experience to ActionTimeStep here?

Reply:
It's not necessary , and _rl_algorithm.train_step require Experience as input